### ISSUE TITLE
[Question] Sequential (per-entity) Training in historical_forecast for Multi-Entity Data Instead of Global Model Training

### ISSUE THREAD

**Issue by LVystorobska** [unit8co/darts#2467]
#### Description
There is an issue with the historical_forecast method in the library when working with multi-entity datasets and models that support global training. 
The method currently processes each entity's data individually (iterates over each TimeSeries object), training and forecasting sequentially for each entity (rewriting the model checkpoint for each new time chunk/entity). 
This contrasts with the expected behavior where the method should train a global model using data from all entities collectively (if this is the input), when selected model supports global training. This behavior is confusing and not explicitly documented.

#### Question
Is the current sequential behavior intended?
If so, it should be clearly specified in the documentation to avoid confusion.

#### Request
I suggest that there should be a parameter/ condition, of course with added modification, thus the method could train a global model using data from all entities (as single epoch iteration, as it is doing now for simple train/test fitting), ensuring that the model benefits from the aggregated information.

Of course, it is possible to implement it on top of darts fitting functionality, but still the current behavior should be mentioned somewhere.

**Comment by tniveej** [comment]
Hello!

Just to note that I have tried this with the `TiDEModel` . And the behaviour in `model.fit()` is indeed 'global'. Where one epoch is when the model goes through the entire list of training Timeseries'. And checkpointing is only done at the end of each validation run. 

Also when you mention that the training should be done 
> using data from all entities collectively

The only difference I would see from collective and sequential training would be that they get shuffled together?

**Comment by LVystorobska** [comment]
Hi!
I mentioned method historical_forecasts not fit, fit works as expected, and thats what I awaited from historical_forecasts;
When historical forecasts running, for example if we set 3 epochs, we have time series for 100 entities and there should be 5 window shifts,  i expect to see in terminal just 5 model runs (fit) where each one is 3 epochs long; while what you are getting now, 5 shifts x 100 entities runs; 
by collectively I just mean together, the same way it is happening with .fit(), and the whole point is to get per window prediction per entity based on the model trained on all entities (for the current time window), while now it will be the prediction based on single entity train.

**Comment by dennisbader** [comment]
Hi @LVystorobska, we have this in our backlog already in #1538. It also explains why it's not a trivial task, and links all related issues.

I'm closing this issue as it's a duplicate of #1538.
